src/main/java/nl/knaw/dans/ttv/core/service/InboxWatcher.java

```diff
@@ -42,51 +40,46 @@ public InboxWatcher(Path path, String datastationName, Callback callback, int in
         this.path = Objects.requireNonNull(path, "InboxWatcher path must not be null");
         this.datastationName = datastationName;
         this.callback = Objects.requireNonNull(callback, "InboxWatcher callback must not be null");
-        this.interval = Objects.requireNonNullElse(interval, 500);
+        this.interval = interval;
     }
 
     @Override
-    public void onFileCreate(File file) {
-        // see if file is a direct descendant of path
-        // if not, ignore it
-        var expected = Path.of(String.valueOf(this.path), file.getName());
-        log.debug("Comparing directories: '{}' vs '{}'", file.toPath(), expected);
+    public void start() throws Exception {
+        log.info("Starting InboxWatcher for path '{}'", this.path);
 
-        if (!file.toPath().equals(expected)) {
-            log.warn("File found in non-root directory, ignoring");
-            return;
+        try {
+            log.info("Starting file alteration monitor for path '{}'", this.path);
+            startFileAlterationMonitor();
+        }
+        catch (IOException e) {
+            log.error(e.getMessage(), e);
+            throw e;
         }
-
-        this.callback.onFileCreate(file, datastationName);
     }
 
     private void startFileAlterationMonitor() throws Exception {
-        FileAlterationObserver observer = new FileAlterationObserver(path.toFile());
+        var observer = new FileAlterationObserver(path.toFile());
         observer.addListener(this);
 
         monitor = new FileAlterationMonitor(this.interval);
-        monitor.addObserver(observer);
         monitor.start();
+        // adding the observer after starting the monitor makes it also include existing files
+        monitor.addObserver(observer);
```

Comment:
- Make a custom FileAlterationObserver that overrides initialize()
- Check what happens if the folder is there, but not readable